frontend/src/pages/studio/home/home-container.tsx

```diff
@@ -1,28 +1,35 @@
-import { DataStatus, IconName } from 'common/enums/enums';
+import { AppRoutes, DataStatus, IconName } from 'common/enums/enums';
 import { FC } from 'common/types/types';
 import { createToastNotification, Loader } from 'components/common/common';
 import { Forbidden, NotFound } from 'components/placeholder-page';
-import { useAppDispatch, useAppSelector, useCallback, useEffect } from 'hooks/hooks';
-import { errorCodes } from 'shared/build';
+import { useAppDispatch, useAppSelector, useCallback, useEffect, useNavigate } from 'hooks/hooks';
+import { errorCodes, StreamPermission } from 'shared/build';
 import { streamActions } from 'store/actions';
 import { StudioStreamContainer } from '../stream/stream-container';
 import { StudioHome } from './home';
 
 const StudioHomeContainer: FC = () => {
   const dispatch = useAppDispatch();
-
-  const { channel, stream, errorCode, error, status } = useAppSelector((state) => ({
+  const navigate = useNavigate();
+  const { channel, stream, errorCode, error, status, streamPermission } = useAppSelector((state) => ({
     stream: state.stream.stream,
     channel: state.stream.channel,
     status: state.stream.status.dataStatus,
     errorCode: state.stream.status.errorCode,
     error: state.stream.status.error,
+    streamPermission: state.auth.streamPermission,
   }));
 
   const handleStartStreaming = useCallback(() => {
     dispatch(streamActions.createStream({ channelId: channel?.id ?? 'ERROR' }));
   }, [dispatch, channel?.id]);
 
+  useEffect(() => {
+    if (streamPermission.streamPermission !== StreamPermission.ALLOWED) {
```

Comment:
we need protect all routes not only home page

Reply:
Done!